lib/theory/src/TheoryObject.hs

```diff
@@ -440,6 +444,23 @@ addLemma l thy = do
     guard (isNothing $ lookupLemma (L.get lName l) thy)
     return $ modify thyItems (++ [LemmaItem l]) thy
 
+-- | Add a new lemma at a specific index. Fails, if a lemma with the same name exists.
+addLemmaAtIndex :: Lemma p -> Int -> Theory sig c r p s -> Maybe (Theory sig c r p s)
+addLemmaAtIndex l i thy = do
+    guard (isNothing $ lookupLemma (L.get lName l) thy)
+    return $ modify thyItems (\ls -> (take i ls) ++ [LemmaItem l] ++ (drop i ls)) thy
```

Comment:
Super minor comment, bout you _could_ use `splitAt` here instead of `take` and `drop`. Let me know whether you think this improves the code.